Lua/setup.lua

```diff
@@ -16,25 +16,36 @@ event.Persist("key char","uksrt", function(c)
 end)
 
 -- Widescreen centering hack
-local function pref(name)
-    return PREFSMAN:GetPreference(name)
-end
-
-if  pref("CenterImageTranslateX") ~= 0 or
-    pref("CenterImageTranslateY") ~= 0 or
-    pref("CenterImageAddWidth") ~= 0 or
-    pref("CenterImageAddHeight") ~= 0 then
-    event.Persist("update","afts sucks", function ()
-        DISPLAY:ChangeCentering(
-            pref("CenterImageTranslateX"),
-            pref("CenterImageTranslateY"),
-            pref("CenterImageAddWidth"),
-            pref("CenterImageAddHeight")
-        )
-    end)
+if FUCK_EXE then
+    local function pref(name)
+        return PREFSMAN:GetPreference(name)
+    end
+    if  pref("CenterImageTranslateX") ~= 0 or
+        pref("CenterImageTranslateY") ~= 0 or
+        pref("CenterImageAddWidth") ~= 0 or
+        pref("CenterImageAddHeight") ~= 0 then
+        event.Persist("update","afts sucks", function()
+            DISPLAY:ChangeCentering(
+                pref("CenterImageTranslateX"),
+                pref("CenterImageTranslateY"),
+                pref("CenterImageAddWidth"),
+                pref("CenterImageAddHeight")
+            )
+        end)
+    end
 end
 
 -- OpenITG EditorShowSongTitle emulation
 if config.EditorShowSongTitle == nil then
     config.EditorShowSongTitle = true
+end
+
+-- Use untruncated SCREEN_WIDTH and SCREEN_CENTER_X for v3.1 below / OpenITG
+if not FUCK_EXE or tonumber(GAMESTATE:GetVersionDate()) < 20200112 then
+    -- really hacky
+    local disp_width = PREFSMAN:GetPreference('DisplayWidth')
+    local disp_height = PREFSMAN:GetPreference('DisplayHeight')
+    --
+    SCREEN_WIDTH = 480 * (disp_width / disp_height)
```

Comment:
This is not good, if the player for example uses anamorphic widescreen (aspect ratio does not match display width/height), then the adjusted SCREEN_WIDTH and SCREEN_CENTER_X will be wrong.

Please change it to use the `DisplayAspectRatio` instead